types.go

```diff
@@ -215,6 +215,32 @@ func IsHostname(target string) bool {
 	return len(r) > 0
 }
 
+// IsGCPProjectId returns true if the target is a GCP Project.
+//
+// The unique, user-assigned id of the project. It 1ust be 6 to 30 lowercase ASCII letters, digits, or hyphens.
+// It must start with a letter. Trailing hyphens are prohibited.
+//
+//	Valid: googleproject
+//	Valid: google-project
+//	Valid: google-project123
+//	Valid: google-123-project
+//	Not valid: googleProject
+//	Not valid: google_project
+//	Not valid: google-project-
+//	Not valid: 123-google-project
+func IsGCPProjectId(target string) bool {
+	if IsHostname(target) {
```

Comment:
I don't think it's necessary to check here if the `target` is a Hostname. It is true that, in the vulcan-api, the functionality that tries to guess the type of and asset won't consider a target to be a GCProject if it is a valid Hostname, but that logic will be in the vulcan-api, here, I think we should just check if the target is a valid CG project id,
It is also true that [here](https://github.com/adevinta/vulcan-types/blob/80ac35591800684591599f6fb6164f4d0e3ff1a8/types.go#L111) in this package, we are doing something similar, but the reason is we wanted to rule out targets that, although are strictly valid docker image identifiers, we wont want to consider them as such.